### ISSUE TITLE
Brainstorming

### ISSUE THREAD

**Issue by sinisarudan** [mprinc/KnAllEdge#139]
(detailed tasks in https://trello.com/c/zYAGnDlU/134-brainstorming)
- finish it to enable wizard-driven management
- provide through enabling **system state** configuration changes that are broadcasted and by that fix receivers environment and possible actions
- setting up distantly for all participants their ontov-filter

**Comment by mprinc** [comment]
MD url: https://github.com/mprinc/KnAllEdge/blob/33aae8ad643096d844553db9ae9dffa5f1f3b08f/src/frontend/app/components/brainstorming/BRAINSTORMING.md

**Comment by sinisarudan** [comment]
By @mprinc  : (Here we talk about toolbar ONLY for participants? and not the additional Form for moderator/brainstorming initiator/**master_of_the_state**)
**generalno**: mislim da treba da se pojavi brainstorming toolbar u dnu ili vrhu koji ce imati kontrolu za siguran rad, a da mapa bude vidljiva, to je po meni bolje nego dijalog, jer tako mozes rijesiti i probleme kreiranja nove ideje, dugmeta, itd, dakle klikne se na neko dugme i onda se kreira ideja NA ZADATOM question nodu, dakle o tome brine brainstorming logika, ali korisnik to vidi na mapi i tamo moze da kuca naziv noda, tagove, property, ... dakle sve isto kao inace ali prosto kultivisano sa brainstorming puzzle. Na taj nacin odrzavamo standardni map pogled i omogucavamo korisnicima da ostanu svjesni njega i shvate sta se tacno desava pri brainstormingu sa mapom, a u isto vrijeme se osjecaju sigurni. Treba vidjeti i tu kako omoguciti te sve map-constraints.

Ja sam vec u **MCM** razvio to za pravila sta moze gdje itd, pa bih to prosirio i onda i ti mozes koristiti, o tome mozemo pricati jos.

**Comment by sinisarudan** [comment]
@mprinc: Yes, we agree, in any case if it is a small dialog or a statusbar, tehy always see what's going on in map, it would not be hidden, and even dialog would hide after creation of the new idea.
But statusbar/toolbar is better anyway.

Just, @mprinc _provide a **placeholder** for it_: some regular one that would just be filled/injected by active component(s)/puzzles, in this case: Brainstorming puzzle

**Comment by mprinc** [comment]
OK, then this is my next task

**Comment by sinisarudan** [comment]
**DONE**
-

-  prevent participant from **deleting BS-Question** or its ancestor!
- finish it to enable wizard-driven management
- ⚠️ ❗  broadcast and update on server change of **Broadcast_privacy**!
- ⚠️ hide BrainstormingPanel upon finishing Brainstorming
and ⚠️  💣  **PREVENT USERS FROM CLOSING** it while Brainstorming, because will not be able to open it again
- ❗ 💣 fix **focusToQuestion**
- ⚠️  setUp in Sharing ideas phase, mustFollowPresenter
- turn off adding ideas in next phases, but if we do add them, then we should fix that IDEA is added and not a KNALLEDGE type of node
![screen shot 2016-07-28 at 11 42 03 pm](https://cloud.githubusercontent.com/assets/1769627/17230530/efe10750-551c-11e6-905e-fa841f0e2bea.png)
